components/server/src/user/user-controller.ts

```diff
@@ -621,7 +620,7 @@ export class UserController {
             return;
         }
 
-        if (this.urlStartsWith(returnToURL, this.config.hostUrl.toString()) || this.urlStartsWith(returnToURL, this.config.brandingConfig.homepage)) {
+        if (this.urlStartsWith(returnToURL, this.config.hostUrl.toString()) || this.urlStartsWith(returnToURL, "https://www.gitpod.io")) {
```

Comment:
I don't know exactly how the attack vector looks like that we want to address with this check, however, this `urlStartsWith(...)` check would also hold for a potentially malicious domain like `https://www.gitpod.io.malicious.com`, right? Thus, you need at least add the trailing slash to the domain or do a proper host check, don't you?


```suggestion
        if (this.urlStartsWith(returnToURL, this.config.hostUrl.toString()) || this.urlStartsWith(returnToURL, "https://www.gitpod.io/")) {
```

Reply:
**fyi**: Cross-linking a relevant change containing dynamic host URL from https://github.com/gitpod-io/gitpod/pull/6528 (Cc @iQQBot) in case there's anything we could replicate here or there.

Reply:
`this.urlStartsWith()` already adds the trailing slash if missing